dot/network/discovery.go

```diff
@@ -185,6 +201,20 @@ func (d *discovery) findPeers(ctx context.Context) {
 	}
 
 	for {
+		defer func() {
```

Comment:
why are we creating a defer function for every loop?

Reply:
removed